activities/activity_nick_grokhowsky.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+"""
+Instructions
+
+- Functions intended to run for each scan
+  need to be named run_xxxxx
+
+- Do not modify the parameters of the run_xxx function
+  unless you know what you are doing
+  see optional parameters:
+  https://github.com/tangible-landscape/grass-tangible-landscape/wiki/Running-analyses-and-developing-workflows#python-workflows
+
+- All gs.run_command/read_command/write_command/parse_command
+  need to be passed env parameter (..., env=env)
+"""
+
+import grass.script as gs
+
+# Function to show significantly high and low pixels 
+def run_highestPixelValues(elev, env, **kwargs):
+	# Create a smooth surface from the elevation LiDAR points
+    gs.run_command(
+        'r.neighbors',
+         input=elev,
+         output='smooth',
+         method='average',
+         flags='c',
+	 env=env
+    )
+    # Calculate univariate statistics
+    stats = gs.parse_command('r.univar', map='smooth', env=env)
+    
+    # Parse the stats library and capture the values for mean and standard deviation
+    for value in stats:
+        if 'mean' in value:
+            mean = float(value[ -7: ])
+        if 'standard deviation' in value:
+            sd = float(value[ -7: ])
+    high = str(mean + sd)
+    low = str(mean - sd)
+
+    # Using map algebra create a new raster map of highest and lowest pixel values versus all others
+    gs.mapcalc('high_values=if(smooth >= ' + high + ')') 
+    gs.mapcalc('new_values=high_values + 1') 
+    gs.mapcalc('low_values=if(smooth > ' + low + ')') 
+    gs.mapcalc('final_values=high_values + low_values') 
```

Comment:
Could you explain what are you trying to do? Please check the raster algebra expressions.

Reply:
I know this looks a little wonky!  I was trying out mapcalc to identify high pixel values and low pixel values.  The logic was to create a binary raster of significantly high values (>95% values) equal to 1 and all other values equal to 0.  Then I added 1 to all pixels, which should make high values 2 and all other values 1.  Next, I created another binary raster of significantly low values (<5%) equal to 0 and all other values equal to 1.  Finally, I added the two output raster images which should produce a raster of  significant high values equal to 3, insignificant values equal to 2, and significant low values equal to 0.  Please let me know if this makes sense.  Also, I am sure there was a better way to do this, but I am not entirely sure of the syntax for mapcalc yet, but working on it.

Reply:
The syntax you want to use is `if(condition, value1, value2)` (value1 when condition is true, value2 when condition is false). If I understood you correctly, you can use a nested loop like this:
```
f"result = if (smooth < {low}, 1, if(smooth > {high}, 3, 2))"
```
(I use Python fstring formating syntax for readability, but concatenating strings is fine)

Also for 95% you need 2 sds.

If you would like to divide it into more bands (1 sd, 2 sd), the nesting would get difficult, so then you would reclassify the raster using r.recode, check the manual page for the syntax.

Reply:
Thanks!  This is super helpful.